src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -1386,4 +1392,91 @@ private boolean checkIfUserIsMemberOfCourseOrganizations(User user, Course cours
+
+        Set<Long> exerciseIdsOfCourse = includedExercises.stream().map(Exercise::getId).collect(Collectors.toSet());
+        CourseManagementDetailViewDTO dto = courseService.getStatsForDetailView(courseId, exerciseIdsOfCourse);
+
+        // Only counting assessments and submissions which are handed in in time
+        long numberOfAssessments = resultRepository.countNumberOfAssessments(exerciseIdsOfCourse).inTime();
+        dto.setCurrentAbsoluteAssessments(numberOfAssessments);
+        long numberOfSubmissions = submissionRepository.countByCourseIdSubmittedBeforeDueDate(courseId)
+                + programmingExerciseRepository.countLegalSubmissionsByCourseIdSubmitted(courseId);
+        dto.setCurrentMaxAssessments(numberOfSubmissions);
+        if (numberOfSubmissions > 0) {
+            dto.setCurrentPercentageAssessments(Math.round(numberOfAssessments * 1000.0 / numberOfSubmissions) / 10.0);
+        }
+        else {
+            dto.setCurrentPercentageAssessments(0.0);
+        }
+
+        // Complaints
+        long numberOfAnsweredComplaints = complaintResponseRepository
+                .countByComplaint_Result_Participation_Exercise_Course_Id_AndComplaint_ComplaintType_AndSubmittedTimeIsNotNull(courseId, ComplaintType.COMPLAINT);
+        dto.setCurrentAbsoluteComplaints(numberOfAnsweredComplaints);
+        long numberOfComplaints = complaintRepository.countByResult_Participation_Exercise_Course_IdAndComplaintType(courseId, ComplaintType.COMPLAINT);
+        dto.setCurrentMaxComplaints(numberOfComplaints);
+        if (numberOfComplaints > 0) {
+            dto.setCurrentPercentageComplaints(Math.round(numberOfAnsweredComplaints * 1000.0 / numberOfComplaints) / 10.0);
+        }
+        else {
+            dto.setCurrentPercentageComplaints(0.0);
+        }
+
+        // More Feedback Requests
+        long numberOfAnsweredFeedbackRequests = complaintResponseRepository
+                .countByComplaint_Result_Participation_Exercise_Course_Id_AndComplaint_ComplaintType_AndSubmittedTimeIsNotNull(courseId, ComplaintType.MORE_FEEDBACK);
+        dto.setCurrentAbsoluteMoreFeedbacks(numberOfAnsweredFeedbackRequests);
+        long numberOfMoreFeedbackRequests = complaintRepository.countByResult_Participation_Exercise_Course_IdAndComplaintType(courseId, ComplaintType.MORE_FEEDBACK);
+        dto.setCurrentMaxMoreFeedbacks(numberOfMoreFeedbackRequests);
+        if (numberOfMoreFeedbackRequests > 0) {
+            dto.setCurrentPercentageMoreFeedbacks(Math.round(numberOfAnsweredFeedbackRequests * 1000.0 / numberOfMoreFeedbackRequests) / 10.0);
+        }
+        else {
+            dto.setCurrentPercentageMoreFeedbacks(0.0);
+        }
+        // Average Student Score
+        dto.setCurrentMaxAverageScore(reachablePoints);
+        dto.setCurrentAbsoluteAverageScore(round((averageScoreForCourse / 100.0) * reachablePoints));
+        if (reachablePoints > 0.0) {
```

Comment:
Is it possible that some edge cases with rounding occur in that line?

Reply:
We assure that neither reachablePoints nor averageScoreForCourse is null (Stream.sum() returns a `double` and I added a check for a possible average score null values earlier in this method). So i could not think of an scenario where this would make problems...